src/sentry/plugins/sentry_mail/models.py

```diff
@@ -289,6 +289,17 @@ def get_digest_subject(self, group, counts, date):
         )
 
     def notify_digest(self, project, digest):
+        user_ids = self.get_send_to(project)
+        result = get_personalized_digests(project.id, digest, user_ids)
+        # TODO(LB): This looks weird. any ponters?
+        if result == ProjectOwnership.Everyone:
```

Comment:
I think in this case it's better to make the result of `get_personalized_digests` return the same under both conditions, even if the digest isn't changed on each loop iteration. This way the caller doesn't have to perform this check and the `ProjectOwnership` bits are fully encapsulated within `get_personalized_digests`.

There's also the problem that you can't use a `return` argument inside a generator:

```
>>> def gen(cond):
...   if cond:
...     return True
...   else:
...     for i in xrange(10):
...       yield i
... 
  File "<stdin>", line 6
SyntaxError: 'return' with argument inside generator
```

You have a couple options here:

1. Do something similar to what you're doing now by checking for the existence of the `ProjectOwnership` and switching based on that, e.g.

```python
def get_personalized_digests(project_id, digest, user_ids):
   if not ProjectOwnership.objects.filter(project_id=project_id).exists():
       for user_id in user_ids:
           yield user, digest
   else:
       # Do what you're already doing for the rest of the function.
```

2. Alternatively, you can pass the set of users that are possible recipients into in `build_events_by_actor`, and replace the `ProjectOwnership.Everyone` result with the set of all recipients (since that is the intersection of "everyone" and the possible recipients.) This way you only have to interact with `ProjectOwnership` directly in one spot. (Personally, I think this option is preferable.)